Classes/Command/DeutscherWetterdienstWarnCellCommand.php

```diff
@@ -0,0 +1,161 @@
+            return Command::SUCCESS;
+        } catch (\Throwable $e) {
+            $this->logger->error(
+                sprintf('Error while updating warn cells: %s', $e->getMessage()),
+                ['exception' => $e],
+            );
+            $output->writeln($e->getMessage());
+            $output->writeln('<error>An error occurred. Check the logs for details.</error>');
+            return Command::FAILURE;
+        }
+    }
+
+    protected function fetchWarnCellData(): ResponseInterface
+    {
+        try {
+            $response = $this->requestFactory->request($this::API_URL);
+            if ($response->getStatusCode() !== 200 || (string)$response->getBody() === '') {
+                $this->logger->log(
+                    LogLevel::ERROR,
+                    WeatherUtility::translate('message.api_response_null', 'deutscherwetterdienst'),
+                );
+                throw new \RuntimeException('Invalid response from API.');
+            }
+
+            return $response;
+        } catch (\Exception $e) {
+            throw new \RuntimeException('Failed to fetch warn cell data: ' . $e->getMessage(), 0, $e);
+        }
+    }
+
+    /**
+     * @return array<int, mixed>
+     */
+    private function parseResponse(ResponseInterface $response): array
+    {
+        $rawRows = explode(PHP_EOL, trim((string)$response->getBody()));
+        array_shift($rawRows); // Remove header row
+
+        $rows = [];
+        foreach ($rawRows as $index => $rawRow) {
+            $fields = str_getcsv($rawRow, ';');
+            if (count($fields) !== 5) {
+                $this->logger->warning(sprintf('Malformed row at line %d: %s', $index + 2, $rawRow));
+                continue;
+            }
+
+            [$warnCellId, $name, $nuts, $shortName, $sign] = $fields;
+            $rows[] = [
+                'warn_cell_id' => $warnCellId,
+                'name' => $name,
+                'nuts' => $nuts,
+                'short_name' => $shortName,
+                'sign' => $sign,
+            ];
+        }
+
+        return $rows;
+    }
+
+    /**
+     * @param array<int, mixed> $rows
+     */
+    private function updateDatabase(array $rows, OutputInterface $output): void
+    {
+        $connection = $this->connectionPool->getConnectionForTable('tx_weather2_domain_model_dwdwarncell');
+
+        $progressBar = new ProgressBar($output, count($rows));
+        $progressBar->start();
+
+        foreach ($rows as $row) {
+            if (!$this->doesRecordExist($connection, $row['warn_cell_id'])) {
+                $this->insertRecord($connection, $row);
+            }
+            $progressBar->advance();
+        }
+
+        $progressBar->finish();
+        $output->writeln('');
+    }
+
+    private function doesRecordExist(Connection $connection, string $warnCellId): bool
+    {
+        $queryBuilder = $connection->createQueryBuilder();
+        $count = $queryBuilder
+            ->count('uid')
```

Comment:
A count cost too much performance. Just select "uid" and check, if value is false or not 0